### PR TITLE
Setup test automation for Java SDK

### PR DESCRIPTION
**Pull request by maria-robobug**
This PR adds the test automation setup for the Java code samples, and adds 2 initial tests to ensure the setup works.
The changes are quite similar to the Go SDK [PR](https://github.com/couchbase/docs-sdk-go/pull/117).

Please note when we compile the example Java classes, I've excluded the ones in here for the moment as I am not sure why they have their own separate `pom.xml` file with version `3.0.10` of the couchbase client:
https://github.com/couchbase/docs-sdk-java/blob/release/3.1/modules/devguide/examples/java/pom.xml
Any historical reasons? I can't see many of them referenced in .adoc files, but let me know if I need to include them.

I am planning to add tests for all other remaining examples in a separate PR so that this changeset doesn't get too big.

### CODE REVIEW

File: modules/pom.xml [hunk 1]
```diff
@@ -1,7 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (osfameron) [inline modules/pom.xml:2]: original formatting was better in this case.

File: modules/test/scripts/init-couchbase/init-buckets.sh [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+set -euo pipefail
+
+CB_USER="${CB_USER:-Administrator}"
+CB_PSWD="${CB_PSWD:-password}"
+
+CB_BUCKET_RAMSIZE="${CB_BUCKET_RAMSIZE:-128}"
+
+echo "couchbase-cli bucket-create travel-sample..."
+/opt/couchbase/bin/couchbase-cli bucket-create \
+-c localhost -u ${CB_USER} -p ${CB_PSWD} \
```
Comment (osfameron) [inline modules/test/scripts/init-couchbase/init-buckets.sh:16]: Missing indentation on these continuation lines looks odd.

Comment (osfameron) [reply modules/test/scripts/init-couchbase/init-buckets.sh:16]: (and below throughout. not sure what our coding standard is here, but I've always seen them indented.)

File: modules/pom.xml (hunk 1 above)
Comment (maria-robobug) [reply modules/pom.xml:2]: Not sure why, my VSCode just decided to change that, let me double check

File: modules/howtos/examples/TransactionsExample.java [hunk 3]
```diff
@@ -547,19 +538,16 @@ static void deferredRollback(byte[] encoded) {
...
     static void configCleanup(byte[] encoded) {
         // tag::config-cleanup[]
-        Transactions transactions = Transactions.create(cluster, TransactionConfigBuilder.create()
-                .cleanupClientAttempts(false)
-                .cleanupLostAttempts(false)
-                .cleanupWindow(Duration.ofSeconds(120))
-                .build());
+        Transactions transactions = Transactions.create(cluster,
+                TransactionConfigBuilder.create().cleanupClientAttempts(false).cleanupLostAttempts(false)
+                        .cleanupWindow(Duration.ofSeconds(120)).build());
```
Comment (osfameron) [inline modules/howtos/examples/TransactionsExample.java:550]: pipeline indendation before looked cleaner

File: modules/test/scripts/init-couchbase/init-buckets.sh (hunk 2 above)
Comment (maria-robobug) [reply modules/test/scripts/init-couchbase/init-buckets.sh:16]: I believe it's my `bash-beautify` VSCode plugin, it formats on save (https://marketplace.visualstudio.com/items?itemName=shakram02.bash-beautify)

Comment (maria-robobug) [reply modules/test/scripts/init-couchbase/init-buckets.sh:16]: It's been handy, but in this particular case it just seems to remove the indent, I'm not really sure what the standard is here myself. Let me switch it off and add an indent.

**Comment by osfameron** [comment]
The "Ensure all classes can compile" commit has rather a lot of reformatting changes that make it hard to review, but looks sensible from a quick scan. Again I'll do a quick run at some point to check it's Hakim-proof :)

**Comment by maria-robobug** [comment]
Regarding the commit, I think the problem is when I make those changes, my java extension (vscjava.vscode-java-pack) also formats the class on save. It's handy as it tells me when there are syntax issues etc... also I'm not sure if these examples have been formatted correctly previously or just differently to what my extension is doing.
https://code.visualstudio.com/docs/languages/java

I dunno if there is a way to automatically set a common style...

**Comment by osfameron** [comment]
Dunno usually prettifiers are either configurable *or* the One True Standard?

If that's the standard, then no worries, but it does make it hard to review. (Doing a reformat commit, as you've done elsewhere is good.)

**Comment by maria-robobug** [comment]
For next time I'll try to maybe cover formatting first(commit separately), then make changes to get the example working or whatnot. That should hopefully make it easier to look at!

File: modules/pom.xml (hunk 1 above)
Comment (maria-robobug) [reply modules/pom.xml:2]: Resolving as this is prettifier related like we discussed below.

File: modules/test/scripts/init-couchbase/init-buckets.sh (hunk 2 above)
Comment (maria-robobug) [reply modules/test/scripts/init-couchbase/init-buckets.sh:16]: I guess this is also following some kind of prettifier standard, so I might just leave it as is

**Pull request merged**